platform/commonUI/edit/src/services/TransactionService.js

```diff
@@ -0,0 +1,118 @@
+/*****************************************************************************
+ * Open MCT Web, Copyright (c) 2014-2015, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT Web is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT Web includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define*/
+define(
+    [],
+    function() {
+        /**
+         * Implements an application-wide transaction state. Once a
+         * transaction is started, calls to PersistenceCapability.persist()
+         * will be deferred until a subsequent call to
+         * TransactionService.commit() is made.
+         *
+         * @param $q
+         * @constructor
+         */
+        function TransactionService($q, dirtyModelCache) {
+            this.$q = $q;
+            this.transaction = false;
+            this.committing = false;
+            this.cache = dirtyModelCache;
+        }
+
+        TransactionService.prototype.startTransaction = function () {
+            if (this.transaction)
+                console.error("Transaction already in progress")
+            this.transaction = true;
+        };
+
+        TransactionService.prototype.isActive = function () {
+            return this.transaction;
+        };
+
+        TransactionService.prototype.isCommitting = function () {
+            return this.committing;
+        };
+
+        /**
+         * All persist calls deferred since the beginning of the transaction
+         * will be committed. Any failures will be reported via a promise
+         * rejection.
+         * @returns {*}
+         */
+        TransactionService.prototype.commit = function () {
+            var self = this;
+                cache = this.cache.get();
+
+            this.committing = true;
+
+            function keyToObject(key) {
+                return cache[key];
+            }
+
+            function objectToPromise(object) {
+                return object.getCapability('persistence').persist();
```

Comment:
I like the simplicity of what you're proposing. This did occur to me as well at the time, but I thought it might be useful to actually be able to track the dirtiness of objects, which is also why the dirty cache is implemented separately. Although we are not using dirtiness for anything other than transactions, this ability might be desirable in future.

On balance, I think your proposal makes sense, and I will refactor it. Dirtiness may be useful in future, but we will cross that bridge when we come to it.